lib/src/supertokens.dart

```diff
@@ -69,4 +94,91 @@ class SuperTokens {
     String? idRefreshToken = await IdRefreshToken.getToken();
     return idRefreshToken != null;
   }
+
+  static Future<void> signOut(Function(Exception?) completionHandler) async {
+    if (!(await doesSessionExist())) {
+      SuperTokens.config!.eventHandler(Eventype.SIGN_OUT);
+      completionHandler(null);
+      return;
+    }
+
+    Uri uri;
+    try {
+      uri = Uri.parse(SuperTokens.signOutUrl);
+    } catch (e) {
+      completionHandler(SuperTokensException(
+          "Please provide a valid apiDomain and apiBasePath"));
+      return;
+    }
+
+    http.Request signOut = http.Request('post', uri);
+    SuperTokens.config!.preAPIHook(APIAction.SIGN_OUT, signOut);
+
+    var resp;
+
+    try {
+      resp = await signOut.send();
+      if (resp.statusCode >= 300) {
+        completionHandler(SuperTokensException(
+            "Sign out failed with response code ${resp.statusCode}"));
+        return;
+      }
+      SuperTokens.config!.postAPIHook(APIAction.SIGN_OUT, signOut, resp);
+    } catch (e) {}
+
+    try {
+      var dataStr = await resp.stream.bytesToString();
+      Map<String, dynamic> data = jsonDecode(dataStr);
+
+      if (data['status'] == 'GENERAL_ERROR') {
+        completionHandler(SuperTokensException(data['message']));
```

Comment:
General Errors need to be a specific error type and not the same as normal errors thrown from the SDK